### PR TITLE
Shutdown aware handler

### PR DESCRIPTION
**Pull request by tanghel**

### CODE REVIEW

File: lerna.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "$schema": "node_modules/lerna/schemas/lerna-schema.json",
   "useWorkspaces": true,
-  "version": "3.6.0",
+  "version": "3.7.1",
```
Comment (bogdan-rosianu) [inline lerna.json:4]: maybe 3.7.0 ?

File: packages/common/src/common/critical-path/shutdown-aware.handler.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import { INestApplication, INestMicroservice } from "@nestjs/common";
+import { ShuttingDownError } from "./shutting-down.error";
+import * as readline from 'readline';
+import { OriginLogger } from "../../utils/origin.logger";
+
+export class ShutdownAwareHandler {
```
Comment (bogdan-rosianu) [inline packages/common/src/common/critical-path/shutdown-aware.handler.ts:6]: do we still need `critical-path.ts` and  `critical-path.handler.ts` ? also, could have renamed the directory from `critical-path` to `shutdown-aware`

Comment (tanghel) [reply packages/common/src/common/critical-path/shutdown-aware.handler.ts]: renamed folder to shutdown-aware, removed critical-path files

File: lerna.json (hunk 1 above)
Comment (tanghel) [reply lerna.json]: done

**Review by bogdan-rosianu — APPROVED** [review]

**Review by gabrielmatei — APPROVED** [review]

**Pull request merged**